packages/next/src/server/lib/trace/tracer.ts

```diff
@@ -135,25 +138,25 @@ class NextTracerImpl implements NextTracer {
   // Trace, wrap implementation is inspired by datadog trace implementation
   // (https://datadoghq.dev/dd-trace-js/interfaces/tracer.html#trace).
   public trace<T>(
-    name: SpanNames,
-    fn: (span: Span, done?: (error?: Error) => any) => Promise<T>
+    type: SpanNames,
+    fn: (span?: Span, done?: (error?: Error) => any) => Promise<T>
   ): Promise<T>
   public trace<T>(
-    name: SpanNames,
-    fn: (span: Span, done?: (error?: Error) => any) => T
+    type: SpanNames,
+    fn: (span?: Span, done?: (error?: Error) => any) => T
   ): T
   public trace<T>(
-    name: SpanNames,
+    type: SpanNames,
     options: TracerSpanOptions,
-    fn: (span: Span, done?: (error?: Error) => any) => Promise<T>
+    fn: (span?: Span, done?: (error?: Error) => any) => Promise<T>
   ): Promise<T>
   public trace<T>(
-    name: SpanNames,
+    type: SpanNames,
     options: TracerSpanOptions,
-    fn: (span: Span, done?: (error?: Error) => any) => T
+    fn: (span?: Span, done?: (error?: Error) => any) => T
   ): T
   public trace<T>(...args: Array<any>) {
-    const [name, fnOrOptions, fnOrEmpty] = args
+    const [type, fnOrOptions, fnOrEmpty] = args
```

Comment:
if you're going ahead with this, might as well rename SpanNames to SpanTypes

Reply:
I didn't realize that :+1:. I will revert that rename.

Reply:
Actually will follow through with the refactor, `name` will be the actually display name for the span.